pyat/at/lattice/utils.py

```diff
@@ -1094,25 +1012,28 @@ def set_tilt(ring: Sequence[Element], tilts, relative=False) -> None:
     """
     tilts = numpy.broadcast_to(tilts, (len(ring),))
     for el, tilt in zip(ring, tilts):
-        tilt_elem(el, tilt, relative=relative)
+        transform_elem(el, tilt=tilt, relative=relative)
 
 
-def set_shift(ring: Sequence[Element], dxs, dzs, relative=False) -> None:
+def set_shift(ring: Sequence[Element], dxs, dys, dzs, relative=False) -> None:
```

Comment:
This should be
```python
def set_shift(ring: Sequence[Element], dxs, dys, dzs=0.0, *, relative=False) -> None:
```
You have to make `dzs` a keyword argument with default 0.0, otherwise you break the compatibility for cases where it is called with only three positional arguments (`set_shift(ring, dxs, dys)`).

In addition, making `relative` keyword only avoids confusion when using 4 positional arguments.